### PR TITLE
[lte][agw] Extending enbConnected to enbState on s1ap_service 

### LINKED ISSUE #4044: Add metric for UE's connected per ENB
For FWA HA we need to know when an ENB has users connected to it. Currently we are using ENB throughput to determine this. This is an issue as throughput can be 0 when all users are in IDLE state. To better determine this we want to a metric for each user connected to each ENB.

### PR DESCRIPTION
**Pull request by ardzoht**
## Summary

- This updates GetEnbConnected to GetEnbState for getting state of S1 connected enodebs with number of UEs connected to each radio
- Updates get_operational_states from enodebd to include ues_connected on the json dict
- Updates orc8r enodeb_state model struct to include new ues_connected property

## Test Plan

- Testing with local orc8r setup and gateway VM, registering unmanaged enb config on Orchestrator and running s1ap tester to simulate enb and 5 UEs connected:

```
{
  "enodeb_configured": false,
  "enodeb_connected": true,
  "fsm_state": "N/A",
  "gps_connected": false,
  "gps_latitude": "N/A",
  "gps_longitude": "N/A",
  "ip_address": "192.168.1.142",
  "mme_connected": true,
  "opstate_enabled": false,
  "ptp_connected": false,
  "reporting_gateway_id": "gw_enb",
  "rf_tx_desired": false,
  "rf_tx_on": false,
  "time_reported": 1608269570650,
  "ues_connected": 5
}
```

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by mpgermano — COMMENTED** [review]
I am conducting some serialization tests locally to ensure the model change is okay. Let's hold off on landing until these are complete.

**Comment by mpgermano** [comment]
> I am conducting some serialization tests locally to ensure the model change is okay. Let's hold off on landing until these are complete.

Everything checked out so this is good on my end.

**Review by ulaskozat — APPROVED** [review]
LGTM. We should consider changing cell ID to eNB ID.

File: lte/protos/s1ap_service.proto
Comment (hcgatewood) [inline lte/protos/s1ap_service.proto]: Prefer golang-style comments (prepend field name), and use proper upper/lowercasing. E.g.

```
// enb_state_map maps { eNB IDs -> # UEs connected }.
map<uint32, uint32> enb_state_map = 1;
```

File: lte/protos/s1ap_service.proto
Comment (hcgatewood) [inline lte/protos/s1ap_service.proto]: Service definition should be at the top of the file, followed by the request+result messages. Aids readability + getting your point across

File: lte/protos/s1ap_service.proto
Comment (hcgatewood) [inline lte/protos/s1ap_service.proto]: 1. Always include comment describing what the rpc does

2. Request and return messages should always be specific to this method, and named as `MethodnameRequest`, `MethodnameResponse` [0]. Here this would be (with consistent casing [1])

```
rpc GetENBState (GetENBStateRequest) returns (GetENBStateResponse) {}
```

[0] https://github.com/uber/prototool/blob/dev/style/README.md#servicesrpcs
[1] https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: lte/gateway/python/magma/enodebd/enodeb_status.py
Comment (hcgatewood) [inline lte/gateway/python/magma/enodebd/enodeb_status.py:348]: Nit for here and elsewhere: [avoid Hungarian notation](https://docs.quantifiedcode.com/python-anti-patterns/readability/putting_type_information_in_a_variable_name.html). Name the variable something semantically meaningful. Then use typing information to handle the rest

E.g. here I like just `enb_statuses`

File: lte/cloud/go/services/lte/obsidian/models/defaults.go
Comment (hcgatewood) [inline lte/cloud/go/services/lte/obsidian/models/defaults.go:85]: This file should be in a separate `test_models` package. Since it's kind of a big change we don't have to do it here. Instead, can you just add a doc comment to the three functions in this file that they're only supposed to be used in tests

File: lte/cloud/go/services/lte/obsidian/models/enodeb_state_swaggergen.go
Comment (hcgatewood) [inline lte/cloud/go/services/lte/obsidian/models/enodeb_state_swaggergen.go]: Will this value get set in a forthcoming PR? Right now it's always gonna be zero

**Review by hcgatewood — APPROVED** [review]
Lgtm. Please address comments before merging

File: lte/gateway/python/magma/enodebd/s1ap_client.py
Comment (pshelar) [inline lte/gateway/python/magma/enodebd/s1ap_client.py]: since return optional dict, we can either return None or {}.

**Review by pshelar — APPROVED** [review]
LGTM

**Pull request merged**
